dataprep/clean/clean_text.py

```diff
@@ -0,0 +1,442 @@
+"""
+Clean a DataFrame column containing text data.
+"""
+import re
+import string
+from functools import partial, update_wrapper
+from typing import Any, Callable, Dict, List, Optional, Set, Union
+from unicodedata import normalize
+
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+
+from ..assets.english_stopwords import english_stopwords
+from .utils import NULL_VALUES, to_dask
+
+REGEX_BRACKETS = {
+    "angle": re.compile(r"(\<)[^<>]*(\>)"),
+    "curly": re.compile(r"(\{)[^{}]*(\})"),
+    "round": re.compile(r"(\()[^()]*(\))"),
+    "square": re.compile(r"(\[)[^\[\]]*(\])"),
+}
+REGEX_DIGITS = re.compile(r"\d+")
+REGEX_DIGITS_BLOCK = re.compile(r"\b\d+\b")
+REGEX_HTML = re.compile(r"<[A-Za-z/][^>]*>|&(?:[a-z0-9]+|#[0-9]{1,6}|#x[0-9a-f]{1,6});")
+REGEX_PUNCTUATION = re.compile(fr"[{re.escape(string.punctuation)}]")
+REGEX_URL = re.compile(r"(?:https?://|www\.)\S+")
+REGEX_WHITESPACE = re.compile(r"[\n\t]|[ ]{2,}")
+
+
+def clean_text(
+    df: Union[pd.DataFrame, dd.DataFrame],
+    column: str,
+    pipeline: Optional[List[Dict[str, Any]]] = None,
+    stopwords: Optional[Set[str]] = None,
+) -> pd.DataFrame:
+    """
+    Clean text data in a DataFrame column.
+
+    Parameters
+    ----------
+        df
+            A pandas or Dask DataFrame containing the data to be cleaned.
+        column
+            The name of the column containing text data.
+        pipeline
+            A list of cleaning functions to be applied to the column. If None,
```

Comment:
One more thing: I think the user might look at the API reference to find out the supported cleaning operations. If you agree, do you think we can list the possible pipeline elements here?

Reply:
Sure, I think that's a good idea. Just to clarify, by "possible pipeline elements" do you mean listing the dictionaries like in `default_text_pipeline()` but for all the supported functions?

Reply:
Yes, that's what I was thinking. But upon further thought, I'm not sure if that's enough for the operations with parameters. E.g., for `replace_bracketed`, should we tell the user we support `"curly"`, `"square"`, ...? I'm not sure how we can do this if we just give a list of dictionaries. Also, we have limited formatting options in the sphinx docs. If you have any other ideas, feel free to pursue them, but another option is to just link to the user guide from this part of the docstring.

Reply:
Yeah, I think linking to the user guide is the best option. I'll make sure to do that after I finish the user guide.

Reply:
Sounds good!